internal/application/ingresses.go

```diff
@@ -0,0 +1,188 @@
+package application
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/epinio/epinio/helpers/randstr"
+	"github.com/epinio/epinio/helpers/tracelog"
+	"github.com/epinio/epinio/internal/names"
+	apierror "github.com/epinio/epinio/pkg/api/core/v1/errors"
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+	"github.com/pkg/errors"
+	networkingv1 "k8s.io/api/networking/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// ListRoutes lists all (currently active) routes for the given application
+// The list is constructed from the actual Ingresses and not from the stored
+// information on the Application Custom Resource.
```

Comment:
Since we can't assume that the caller of this method cares about mismatches, I think that validation deserves and dedicated method (`func RoutesInSync() bool` or something). Interested parties would call that explicitly. I'll leave this for the future since currently, any such mismatch can't (shouldn't) occur unless someone tampers with Epinio resources outside the Epinio workflow.